otus-02/src/otus_02/homework/lib.clj

```diff
@@ -0,0 +1,17 @@
+(ns otus-02.homework.lib
+  (:require [clojure.string :as string]))
+
+(defn- is-alpha?
+  [c]
+  (let [i (int c)]
+    (cond (and (> i 96) (< i 123)) true
+          (and (> i 64) (< i 91)) true
+          :else false)))
+
+(defn normalize
+  [s]
+  (->> s
+       string/lower-case
+       vec
+       (filterv is-alpha?)
```

Comment:
Once you call `apply str`, then you don't need force all in memory as vector, just lazy `filter`